test/parallel/test-http-hostname-typechecking.js

```diff
@@ -18,10 +18,16 @@ vals.forEach((v) => {
   assert.throws(() => http.request({host: v}), errCheck('host'));
 });
 
+const reqs = [];
 // These values are OK and should not throw synchronously
 ['', undefined, null].forEach((v) => {
   assert.doesNotThrow(() => {
-    http.request({hostname: v}).on('error', common.noop);
-    http.request({host: v}).on('error', common.noop);
+    reqs.push(http.request({hostname: v}).on('error', common.noop));
```

Comment:
@lpinca See @jasnell’s comment in https://github.com/nodejs/node/pull/12494#discussion_r112709495 … I tried to keep this bit of the test just testing what it’s supposed to. Personally, I would totally be fine with having `.end()` here.